doc/adrs/009_adopt_a_proper_versioning_strategy.adoc

```diff
@@ -0,0 +1,64 @@
+= ADR-009 - Adopt a proper versioning strategy
+
+== Context
+
+We are currently building Sirius Components and Sirius Web.
+They are two relatively new projects and as such, they can change a lot.
+We do not want to set in stone our APIs yet as we want to make changes freely for now.
+We want to iterate quickly yet we need to have a proper versioning strategy to let members of our community update their applications more easily.
+
+Today, every commit of Sirius Components create a new `0.1.0-SNAPSHOT` deployment of the backend and a `0.1.31.COMMIT_ID` deployment of the frontend.
+This creates multiple issues:
+
+- The versions of the frontend and the backend are not coherent
+- We can't easily detect a new deployment of the backend and as such consuming the backend is very complex
+
+Our current development process make us release a new version of Sirius Components and Sirius Web almost every six weeks (plus two "cooldown" weeks).
+We want to see this information in the version number.
+
+
+== Decision
+
+We will use the same version number for both the frontend and the backend, starting with `0.1.32` with this commit.
+We will upgrade this number in two different ways in two situations:
+
+
+=== Upgrade the patch number (0.1.32 to 0.1.33)
+
+We will upgrade the patch number, when we need it to let members of our community consume a new intermediary release.
+There are multiple applications being built on top of Sirius Components.
+When some members of our community needs a new intermediary release to upgrade their application, we will simply upgrade the patch number.
+For example, if in Sirius Components, I create a PR with 3 commits breaking some APIs and if I need to upgrade Sirius Web to use the result of those commits, I can create a fourth commit in the PR upgrading the patch number.
+This way, I can then upgrade Sirius Web with the new version of Sirius Components.
+
+
+=== Upgrade the minor number (0.1.33 to 0.2.0)
+
+When an iteration is done, we need to create a new version of Sirius Components (and of Sirius Web).
+To identify that we have switched to a brand new iteration, we will use the minor number for now.
+We may change that when we will switch to stable APIs that we won't break at a moment's notice.
+
+So after each iteration, we will move from a minor version to the next such as from `0.1.x` to `0.2.0` and then from `0.2.x` to `0.3.0`.D
+uring an iteration, we will make an undetermined number of releases such as `0.1.31`, `0.1.32`, `0.1.33`, `0.1.34` and so on.
+Using a minor number for an iteration will give us the ability to assing Github issues to a specific milestone such as `0.2.0` for the first milestone of 2021.
+
+
+== Status
+
+Accepted.
+
+
+== Consequences
+
+The build will have to be updated to stop deploying new artifacts after each commit and only do so after a commit has been tagged.
```

Comment:
The fact that we *tag* (in the git sense) version on each version bump (inlucing patch number) is not mentioned above (I think).

Reply:
Yes indeed, since we need a way to tell the build that this time we have to deploy the result, I figured tagging was less annoying than creating a new Github release, I'll mention it.